### PR TITLE
feat: support Modal.config for prefixCls

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/25586
close https://github.com/ant-design/ant-design/issues/22886

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | same as #25613  |
| 🇨🇳 Chinese |  same as #25613  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/message/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-modal-config/components/message/index.en-US.md)
[View rendered components/message/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-modal-config/components/message/index.zh-CN.md)
[View rendered components/modal/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-modal-config/components/modal/index.en-US.md)
[View rendered components/modal/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-modal-config/components/modal/index.zh-CN.md)
[View rendered docs/react/faq.en-US.md](https://github.com/ant-design/ant-design/blob/feat-modal-config/docs/react/faq.en-US.md)
[View rendered docs/react/faq.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-modal-config/docs/react/faq.zh-CN.md)

### CODE REVIEW

File: components/message/index.zh-CN.md [hunk 1]
```diff
@@ -89,3 +90,4 @@ message.config({
 | maxCount | 最大显示数, 超过限制时，最早的消息会被自动关闭 | number | - |
 | top | 消息距离顶部的位置 | number | 24 |
 | rtl | 是否开启 RTL 模式 | boolean | false |
+| prefixCls | 消息节点的 className 前缀 | string | 'ant-message' |
```
Comment (zombieJ) [inline components/message/index.zh-CN.md:93]: 版本信息

**Review by zombieJ — APPROVED** [review]

**Review by fireairforce — APPROVED** [review]

File: components/modal/confirm.tsx [hunk 2]
```diff
@@ -59,6 +68,7 @@ export default function confirm(config: ModalFuncProps) {
       ReactDOM.render(
         <ConfirmDialog
           {...props}
+          prefixCls={getPrefixCls(prefixCls)}
```
Comment (Geczy) [inline components/modal/confirm.tsx:71]: `prefixCls={prefixCls || 'ant-modal'}` would be better

File: components/message/index.en-US.md [hunk 3]
```diff
@@ -88,3 +89,4 @@ message.config({
 | maxCount | Max message show, drop oldest if exceed limit | number | - |
 | top | Distance from top | number | 24 |
 | rtl | Whether to enable RTL mode | boolean | false |
+| prefixCls | prefix className of message node | string | 'ant-message' |
```
Comment (xrkffgg) [inline components/message/index.en-US.md:92]:
```suggestion
| prefixCls | prefix className of message node | string | `ant-message` |
```

File: components/message/index.zh-CN.md (hunk 1 above)
Comment (xrkffgg) [inline components/message/index.zh-CN.md:93]:
```suggestion
| prefixCls | 消息节点的 className 前缀 | string | `ant-message` |
```

File: components/message/index.en-US.md (hunk 3 above)
Comment (xrkffgg) [reply components/message/index.en-US.md:92]: Version

**Review by xrkffgg — CHANGES_REQUESTED** [review]

**Comment by xrkffgg** [comment]
我直接来改了吧

**Review by xrkffgg — APPROVED** [review]

File: components/message/index.en-US.md [hunk 4]
```diff
@@ -78,13 +78,15 @@ message.config({
...
-| top | Distance from top | number | 24 |
-| rtl | Whether to enable RTL mode | boolean | false |
+| Argument | Description | Type | Default | Version |
+| --- | --- | --- | --- | --- |
+| duration | Time before auto-dismiss, in seconds | number | 1.5 |  |
+| getContainer | Return the mount node for Message | () => HTMLElement | () => document.body |  |
+| maxCount | Max message show, drop oldest if exceed limit | number | - |  |
+| top | Distance from top | number | 24 |  |
+| rtl | Whether to enable RTL mode | boolean | false |  |
+| prefixCls | prefix className of message node | string | `ant-message` | 4.5.0 |
```
Comment (xrkffgg) [inline components/message/index.en-US.md:92]:
```suggestion
| prefixCls | The prefix className of message node | string | `ant-message` | 4.5.0 |
```

File: components/modal/ConfirmDialog.tsx [hunk 5]
```diff
@@ -38,7 +39,6 @@ const ConfirmDialog = (props: ConfirmDialogProps) => {
 
   // 支持传入{ icon: null }来隐藏`Modal.confirm`默认的Icon
   const okType = props.okType || 'primary';
-  const prefixCls = props.prefixCls || 'ant-modal';
```
Comment (zqjimlove) [inline components/modal/ConfirmDialog.tsx:41]: 为什么要删除这行，现在导致所有 Modal.comfrim 的弹窗都有样式问题

**Pull request merged**